### PR TITLE
Topic Lock

### PR DESCRIPTION
**Pull request by esmaoui**

### CODE REVIEW

File: commons/src/main/java/org/exoplatform/platform/qa/ui/commons/Base.java [hunk 1]
```diff
@@ -85,4 +85,14 @@ public void afterEach() {
...
+<<<<<<< 7027050bad1530600a83b7c9b87c993ec7701f7d
+=======
+
+ /* @AfterEach
+  public void close() {
+    if (!title().equals("Login")) {
+      Selenide.close();
+    }
+  }*/
+>>>>>>> Button Quote should not be visible
```
Comment (thomasdelhomenie) [inline commons/src/main/java/org/exoplatform/platform/qa/ui/commons/Base.java]: To delete

File: platform/src/test/java/org/exoplatform/platform/qa/ui/platform/forum/ForumTopicLockTestIT.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        manageLogInOut = new ManageLogInOut(this);
+        if ($(ELEMENT_SKIP_BUTTON).is(Condition.exist)) {
+            $(ELEMENT_SKIP_BUTTON).click();
+        }
+        manageLogInOut.signInCas(username, password);
+    }
+
+
+    /*
+       bug FORUM-1317
```
Comment (thomasdelhomenie) [inline platform/src/test/java/org/exoplatform/platform/qa/ui/platform/forum/ForumTopicLockTestIT.java]: To be moved in an annotation "@Tag"

File: commons/src/main/java/org/exoplatform/platform/qa/ui/commons/Base.java (hunk 1 above)
Comment (aboughdiri) [reply commons/src/main/java/org/exoplatform/platform/qa/ui/commons/Base.java]: done

File: platform/src/test/java/org/exoplatform/platform/qa/ui/platform/forum/ForumTopicLockTestIT.java (hunk 2 above)
Comment (aboughdiri) [reply platform/src/test/java/org/exoplatform/platform/qa/ui/platform/forum/ForumTopicLockTestIT.java]: corrected

File: platform/src/test/java/org/exoplatform/platform/qa/ui/platform/forum/ForumTopicLockTestIT.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import static org.exoplatform.platform.qa.ui.selenium.testbase.LocatorTestBase.ELEMENT_SKIP_BUTTON;
+
+/**
+ * Created by esmaoui on 16/03/2018.
+ */
+
+
+@Tag("sniff")
+@Tag("forum")
+public class ForumTopicLockTestIT extends Base {
```
Comment (ilyeskallala) [inline platform/src/test/java/org/exoplatform/platform/qa/ui/platform/forum/ForumTopicLockTestIT.java]: You can add this TC in platform/forum/ForumTopicTestIT. Should not create a class for each TC.

File: platform/src/test/java/org/exoplatform/platform/qa/ui/platform/forum/ForumTopicLockTestIT.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        forumCategoryManagement.addCategorySimple(category, "", desc);
+        info("Add a forum in the category");
+        forumForumManagement.addForumSimple(forum, "", desc);
+        info("Add and go to a topic in the forums");
+        forumForumManagement.goToStartTopic();
+        forumTopicManagement.startTopic(topic, topic, "", "");
+        info("Go to a topic");
+        forumHomePage.goToTopic(topic);
+        $(ELEMENT_TOPIC_MORE_ACTION).click();
+        $(ELEMENT_TOPIC_LOCK).click();
```
Comment (ilyeskallala) [inline platform/src/test/java/org/exoplatform/platform/qa/ui/platform/forum/ForumTopicLockTestIT.java]: you can use this function to lock the topic (forumTopicManagement.selectItemMoreActionMenuTopic(specifMoreActionMenuTopic.LOCK);), should not duplicate code

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/forum/ForumLocator.java [hunk 5]
```diff
@@ -891,4 +891,11 @@
   public final By                     ELEMENT_MOVE_TOPIC_POPUP                           = By.id("UIForumPopupWindow");
 
   public static final SelenideElement ELEMENT_BUTTON_SUBMIT_POST= $(byXpath("//*[@id=\"UIPostForm\"]/div[3]/button[1]"));
+
+
+  public static final SelenideElement ELEMENT_TOPIC_MORE_ACTION= $(byXpath("//*[@id=\"UITopicDetail\"]/div[4]/div[1]/ul/li[3]/div/div[1]"));
+
+  public static final SelenideElement ELEMENT_TOPIC_LOCK= $(byXpath("//*[@id=\"UITopicDetail\"]/div[4]/div[1]/ul/li[3]/div/ul/li[5]/a"));
```
Comment (ilyeskallala) [inline selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/forum/ForumLocator.java]: 1.locator already exist (ELEMENT_LOCK_TOPIC) 
2. This locator should be deleted after using the function (forumTopicManagement.selectItemMoreActionMenuTopic(specifMoreActionMenuTopic.LOCK);)

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/forum/ForumLocator.java [hunk 6]
```diff
@@ -891,4 +891,11 @@
   public final By                     ELEMENT_MOVE_TOPIC_POPUP                           = By.id("UIForumPopupWindow");
 
   public static final SelenideElement ELEMENT_BUTTON_SUBMIT_POST= $(byXpath("//*[@id=\"UIPostForm\"]/div[3]/button[1]"));
+
+
+  public static final SelenideElement ELEMENT_TOPIC_MORE_ACTION= $(byXpath("//*[@id=\"UITopicDetail\"]/div[4]/div[1]/ul/li[3]/div/div[1]"));
```
Comment (ilyeskallala) [inline selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/forum/ForumLocator.java]: 1.locator already exist (ELEMENT_MORE_ACTION) 
2. This locator should be deleted after using the function (forumTopicManagement.selectItemMoreActionMenuTopic(specifMoreActionMenuTopic.LOCK);)

File: selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/forum/ForumLocator.java [hunk 7]
```diff
@@ -891,4 +891,11 @@
...
+
+  public static final SelenideElement ELEMENT_TOPIC_LOCK= $(byXpath("//*[@id=\"UITopicDetail\"]/div[4]/div[1]/ul/li[3]/div/ul/li[5]/a"));
+
+  public static final SelenideElement ELEMENT_TOPIC_QUOTE= $(byXpath("//*[@id=\"QuoteId01534136cea60186faec9\"]"));
```
Comment (ilyeskallala) [inline selenium/src/main/java/org/exoplatform/platform/qa/ui/selenium/locator/forum/ForumLocator.java]: It is a dynamique locator, we can't use it.

**Pull request merged**